### DISCUSSION TITLE
Can I use Netlify for umami?

### DISCUSSION THREAD

**Discussion by LeoSeyers** [umami-software/umami#737] [Q&A]
Hello there,

I'm trying to figure out if it's possible to run into a static build with Netlify since there is no much documentation out there.

I'm not really familiar with NextJS but I've edit package.json to have a static export  ( `"build-app": "next build && next export"` ) that can be then deploy with netlify-cli ( `out` folder). In addition to that I've added the .env variables into the Netlify project dashboard.

Somehow it doesn't work – I do have something (see https://demo.studio-scale.com/netlify-umami-fail.jpg) but I'm not sure that's the way to do it and it cannot connect to the database. Also I'm wondering if it's even possible to secure the credential in such static frontend stack?

Anyway Umami looks very good I may try Vercel instead at some point but any hints so far would be much appreciated :)

**Comment by mikecao** [comment]
@LeoSeyers I have not been able to figure out how to run the app on Netlify. I'm sure it's possible since they claim to be able to host Next.js apps. But last time I tried I couldn't get it to work.

**Reply by davidlag0** [reply]
@mikecao Hi, I just deployed the app to Netlify. I followed the instructions on umami's website (https://umami.is/docs/running-on-netlify). I first had some database issues but it was my fault and with the help of discussions here, I was able to resolve these issues. There is one issue that is left for which I have no solution and I hope the fix is simple and perhaps known already. When I visit the site I configured, it makes a request, as it should, to umami but the logs of the Netlify Function "___netlify-handler" show this message:

`
Aug 24, 12:55:17 AM: 6fe8bf16 ERROR  [Error: ENOENT: no such file or directory, open '/var/task/node_modules/.geo/GeoLite2-Country.mmdb'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/var/task/node_modules/.geo/GeoLite2-Country.mmdb'
}
`

and umami does not save any data in the database. I haven't looked at umami's code so I assume the data must be saved in the database at a further step past this point because this file seems to help to geolocate the IP address.

Would you know how to resolve this problem?

Thank you very much!

**Reply by davidlag0** [reply]
@mikecao I'm sorry, I didn't do my homework correctly - I didn't think about searching open issues and pull requests before now. I see there is an open issue for a bug with Netlify: https://github.com/umami-software/umami/issues/1413
